simvue/filter.py

```diff
@@ -0,0 +1,231 @@
+import abc
+import enum
+import sys
+import typing
+
+if sys.version_info < (3, 11):
+    from typing_extensions import Self
+else:
+    from typing import Self
+
+class Status(str, enum.Enum):
+    Created = "created"
+    Running = "running"
+    Completed = "completed"
+    Lost = "lost"
+    Terminated = "terminated"
+    Failed = "failed"
+
+
+class Time(str, enum.Enum):
```

Comment:
```suggestion
class _Time(str, enum.Enum):
```

I think this can be made private/internal, along with `System` and `RestAPIFilter`.